app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ActiveRecord::Base
+  has_many :posts, class_name: 'Post', foreign_key: 'AuthorId'
+  has_many :comments, class_name: 'Comment', foreign_key: 'AuthorId'
+  has_many :likes, class_name: 'Like', foreign_key: 'AuthorId'
+
+  def recent_post
+    posts.order(create_at: :desc).limit(3)
+  end
```

Comment:
- [ SUGGESTION ] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private
EXAMPLE
```ruby
 # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

**You can do the same thing for other models aswell**

Reply:
Thanks a lot for these recommendations bro.